config/config.go

```diff
@@ -1167,13 +1172,16 @@ type StorageConfig struct {
 	// required for `/block_results` RPC queries, and to reindex events in the
 	// command-line tool.
 	DiscardABCIResponses bool `mapstructure:"discard_abci_responses"`
```

Comment:
It can still make sense, at least for a first release of this API. To indicate pruning of ABCI results atm, an operator has to set up a data companion to be attached to the API. And if they do not want to do it at the moment, they can leverage this config flag and avoid storing data they do not want.

Reply:
I see. Makes sense.
Have you folks thought about the possibility of using this pruning mechanism with a noop companion? Think of validators, that would a good fit wouldn't it? Anyway, this is beyond the scope of this PR. But just something to think about.